sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -2459,6 +2459,26 @@ object functions {
     StringTrimLeft(e.expr, Literal(trimString))
   }
 
+  /**
+    * Extracts a part from a URL.
+    *
+    * @group string_funcs
+    * @since 2.4.0
+    */
+  def parse_url(url: Column, partToExtract: String): Column = withExpr {
```

Comment:
When adding the functions into `object functions`, we need to check how useful they are.

Reply:
This PR is created after this StackOverflow question: https://stackoverflow.com/questions/52041342/how-to-parse-url-in-spark-sqlscala/52043771

I'm not sure how often it is used, however most of functions are available in functions object to make Scala and SQL interfaces similar. If you think it's useless - please let me know, I'll just close this PR

Reply:
In long term, I would suggest method that return handler for any registered function. So that you can write: SqlFunction something = spark.(...?).getFunction("parse_url"). Now `spark.udf.register` returns a handler for UDF, something similar for getting any kind of registered function may be helpful. However, it's a lot more work, so now I've just proposed to add this function :)

Reply:
I like this idea too

Reply:
Ok, tomorrow I will create a Jira and start working on it. Thanks for your comments! :)

Reply:
Can't we just use `expr` instead?

Reply:
@HyukjinKwon Thanks for the suggestion, however now users are complaining about stringly-typed system in Spark, there are libs like Frameless from Typelevel to archieve a bit more type safety. `expr` is springly-typed, while functions in `functions` object or accessed via `UserDefinedFunction` are a bit more type safe.

Reply:
I mean, 

> I would suggest method that return handler for any registered function. So that you can write: SqlFunction something = spark.(...?).getFunction("parse_url")

Can this support strongly typed one?

Reply:
@HyukjinKwon I see now. Yeah, wrapping in  the `Column` will be necessary, at least no string concatenation will be required